src/App.jsx

```diff
@@ -13,15 +13,115 @@ function App() {
     const [todoList, setTodoList] = useState([]);
     const [isLoading, setIsLoading] = useState(true);
 
-    useEffect(() => {
-        new Promise((resolve, reject) => {
-            setTimeout(() => {
-                resolve({ data: { todoList: [] } });
-            }, 2000);
-        }).then((result) => {
-            setTodoList(result.data.todoList);
+    // Get tasks from AirTable
+    const fetchData = async () => {
+        const apiURL = `https://api.airtable.com/v0/${
+            import.meta.env.VITE_AIRTABLE_BASE_ID
+        }/${import.meta.env.VITE_TABLE_NAME}`;
+        const options = {
+            method: "GET",
+            headers: {
+                Authorization: `Bearer ${
+                    import.meta.env.VITE_AIRTABLE_API_TOKEN
+                }`,
+            },
+        };
+        try {
+            const response = await fetch(apiURL, options);
+
+            if (!response.ok) {
+                const message = `Error has ocurred: ${response.status}`;
+                throw new Error(message);
+            }
+            const data = await response.json();
+            const todos = data.records.map((todo) => {
+                const newTodo = {
+                    id: todo.id,
+                    title: todo.fields.title,
+                };
+                return newTodo;
+            });
+            setTodoList(todos);
             setIsLoading(false);
-        });
+        } catch (error) {
+            console.log(error.message);
+            return null;
```

Comment:
This function should not return a value, and right now this function will only return null if you catch an error (if not, it will return nothing). Your end result is that it sets the values in state, so we don't need to return anything from the function.